diff --git a/supervisor/datatypes.py b/supervisor/datatypes.py
index 1b3ea26..bd97eb1 100644
--- a/supervisor/dtatypes.py
+++ b/supervisor/datatypes.py
@@ -71,7 +71,7 @@ def dict_of_key_value_pairs(arg):
     """ parse KEY=val,KEY2=val2 into {'KEY':'val', 'KEY2':'val2'}
         Quotes can be used to allow commas in the value
     """
-    lexer = shlex.shlex(arg)
+    lexer = shlex.shlex(arg, posix=True)
     lexer.wordchars += '/.+-():'
 
     tokens = list(lexer)
@@ -83,7 +83,7 @@ def dict_of_key_value_pairs(arg):
         k_eq_v = tokens[i:i+3]
         if len(k_eq_v) != 3 or k_eq_v[1] != '=':
             raise ValueError("Unexpected end of key/value pairs")
-        D[k_eq_v[0]] = k_eq_v[2].strip('\'"')
+        D[k_eq_v[0]] = k_eq_v[2]
         i += 4
     return D
 
